src/SignalR/clients/java/signalr/core/src/main/java/com/microsoft/signalr/HubConnection.java

```diff
@@ -1431,12 +1434,7 @@ public void handleHandshake(ByteBuffer payload) {
         public void timeoutHandshakeResponse(long timeout, TimeUnit unit) {
             handshakeTimeout = Executors.newSingleThreadScheduledExecutor();
             handshakeTimeout.schedule(() -> {
```

Comment:
I'm curious why this uses an executor to implement the timeout instead of using RxJava's timeout operator? 
`connectionState.transport.send(handshake).timeout(..., ...)`

Reply:
Probably implemented before RxJava was used.

Keeping it as-is so it's easier to backport to 5.0